doc/api/test.md

```diff
@@ -1622,6 +1677,54 @@ describe('tests', async () => {
 });
 ```
 
+## `config`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+An object whose methods are used to configure global test runner settings in the
+current process. It is possible to apply the same configuration to all files by
+placing common configuration code in a module preloaded with `--require` or
+`--import`.
```

Comment:
Can this be split out of this PR? It’s come up a few times where people have wanted config files for loaders, for the test runner, for policies and probably other things. I think we should have a single solution for configuring Node via a config file that applies to all of Node’s flags/settings, and not have config files specific to subsystems.

I understand that this isn’t a flag, that this is something specific to the test runner and imported from `node:test`, but the existence of this might constrain the potential design of an all-Node config solution. For example I can imagine something akin to `--env-file` / `util.parseEnv`, like `--config` / `util.parseConfig`, and if we create `util.parseConfig` then we probably shouldn’t also have `node:test.config`.

Reply:
Where would you like `setResolveSnapshotPath()` and `setDefaultSnapshotSerializers()` to be imported from?

Reply:
The `config` object was initially named `snapshot` when I created this PR. I only renamed it to `config` because it seemed like a more general namespace for such methods.

Reply:
It seems like your issue is strictly with the name `config`. I'm happy to rename it to whatever.

Reply:
> I understand that this isn’t a flag, that this is something specific to the test runner and imported from node:test, but the existence of this might constrain the potential design of an all-Node config solution.

That would not be the case here FYI. There is some order of precedence for config between environment variables, command line flags, the hypothetical Node config file, and local configuration. This would be a local configuration that would only override more global configuration.

Reply:
> Where would you like `setResolveSnapshotPath()` and `setDefaultSnapshotSerializers()` to be imported from?

Could you just do `import { setResolveSnapshotPath, setDefaultSnapshotSerializers } from 'node:test'`? Otherwise renaming `config` to `snapshot`, assuming that `snapshot` will contain only snapshot-related methods, is fine.

Yes my objection is to a new API that’s essentially “here’s a way to define broad swaths of configuration” but it’s not _all_ of Node’s configuration. I think we should have the latter, and having miniature versions for subsets of Node will get in the way of creating the bigger thing.